### PR TITLE
community[minor]: Make "IN-filters" possible for PgVector Metadata

### PR DESCRIPTION
**Pull request by aman-shrivastava95**
This change enables IN-filters when doing a similarity search on the Pgvector Vectorstore.

This makes retrieving documents based on metadata filters much more flexible. Currently, only a single value can be checked, this allows for multiple values to be checked against a metadata property.

As for example this Feature Request discussion on the Python package shows, this seems to be possible without being documented in the Python package: https://github.com/langchain-ai/langchain/issues/9726

A similar change is present on the vercel postgres vectorstore
https://github.com/langchain-ai/langchainjs/pull/3040

### CODE REVIEW

File: libs/langchain-community/src/vectorstores/pgvector.ts [hunk 1]
```diff
@@ -408,25 +408,55 @@ export class PGVectorStore extends VectorStore {
...
     if (collectionId) {
       parameters.push(collectionId);
     }
 
+    let paramCount = parameters.length;
+    for (const [key, value] of Object.entries(_filter)) {
+      if (typeof value === "object" && value !== null) {
+        const _value: { [key: string]: any } = value as { [key: string]: any };
+        const currentParamCount = paramCount;
+        const placeholders = _value.in
```
Comment (bracesproul) [inline libs/langchain-community/src/vectorstores/pgvector.ts:430]: If `in` is undefined, placeholders will be undefined too?

File: libs/langchain-community/src/vectorstores/pgvector.ts [hunk 2]
```diff
@@ -408,25 +408,55 @@ export class PGVectorStore extends VectorStore {
...
+    const whereClauses = [];
+
     if (collectionId) {
       parameters.push(collectionId);
     }
 
+    let paramCount = parameters.length;
+    for (const [key, value] of Object.entries(_filter)) {
+      if (typeof value === "object" && value !== null) {
+        const _value: { [key: string]: any } = value as { [key: string]: any };
```
[the remark this answers is not in the document]
Comment (bracesproul) [reply libs/langchain-community/src/vectorstores/pgvector.ts:428]: Also, why is `value` prefixed with an underscore?

**Review by bracesproul — COMMENTED** [review]
Thanks for adding this! I left a few comments. Also, could you please write a test for this?

Thanks again!

**Comment by jacoblee93** [comment]
Thank you, sorry for the delay!

**Pull request merged**